src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -104,13 +104,6 @@ CHIP_ERROR NrfWiFiDriver::CommitConfiguration()
 
 CHIP_ERROR NrfWiFiDriver::RevertConfiguration()
 {
-    // Disconnection should happen automatically when WiFiManager::Connect() is called.
-    // Here we do it also explicitly to ping the Connectivity Manager which
-    // will send the disconnection event as a result.
-    // TODO: refactor when the callback-based supplicant API is incorporated
-    WiFiManager::Instance().DisconnectStation();
```

Comment:
Will the disconnection event be generated after removing this?

Reply:
It will not (actually the reason is removing the line below). But I deleted these 2 lines on purpose as DisconnectStation() call makes the wpa_supplicant thread throw a bus fault in this context (WiFi team will look at that). I mean... we can still trigger the disconnect event without direct call to disconnect but in this case we will not be sure weather or not the station really disconnects (there would have to be a corresponding event captured from supplicant first).

Also I am not sure what is the actual use case of disconnect event, we do not capture it anywhere. Just in case we really need it, it can be added once again in clean manner when the new API is delivered.